src/promptflow/promptflow/_utils/exception_utils.py

```diff
@@ -255,6 +248,53 @@ def error_code_recursed(self):
 
         return current_error
 
+    def to_dict(self, *, include_debug_info=False):
+        """Return a dict representation of the exception.
+
+        This dict specification corresponds to the specification of the Microsoft API Guidelines:
+        https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#7102-error-condition-responses
+
+        Note that this dict representation the "error" field in the response body of the API.
+        The whole error response is then populated in another place outside of this class.
+        """
+        if isinstance(self._ex, JsonSerializedPromptflowException):
+            return self._ex.to_dict(include_debug_info=include_debug_info)
+
+        # Otherwise, return general dict representation of the exception.
+        result = {"message": str(self._ex), "messageFormat": "", "messageParameters": {}}
```

Comment:
Was this line auto-formatted by pre-commit?

Reply:
Yes..